packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,285 @@
+     */
+    public override keydownHandler(event: KeyboardEvent): boolean {
+        if (!this.open) {
+            return false;
+        }
+        switch (event.key) {
+            case keyTab:
+            case keyEnter: {
+                // When there is no filter options, the code is returned.
+                if (this.filteredOptions.length === 0) {
+                    return false;
+                }
+                this.activateMention({
+                    href: this.firstSelectedOption.value,
+                    displayName: this.firstSelectedOption.text
+                } as MentionDetail);
+                return true;
+            }
+            case keyEscape: {
+                this.setOpen(false);
+                return false;
+            }
+            case keyArrowDown:
+            case keyArrowUp: {
+                super.keydownHandler(event);
+                this.scrollOptionIntoView();
+                return false;
+            }
+            default: {
+                super.keydownHandler(event);
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Filter available options by filter value.
+     * The method is defined based on the `Combobox.filterOptions` implementation.
+     *
+     * @internal
+     */
+    public filterOptions(): void {
+        if (!this.filter) {
+            this.filteredOptions = this._options;
+        } else {
+            const normalizedFilter = normalizeString(this.filter);
+            this.filteredOptions = this._options.filter(o => normalizeString(o.text).includes(normalizedFilter));
+        }
+
+        this._options.forEach(o => {
+            o.hidden = !this.filteredOptions.includes(o);
+        });
+    }
+
+    /**
+     * Synchronize the form-associated proxy and update the value property of the element.
+     *
+     * @param prev - the previous collection of slotted option elements
+     * @param next - the next collection of slotted option elements
+     *
+     * @internal
+     */
+    public override slottedOptionsChanged(prev: Element[] | undefined, next: Element[]): void {
+        super.slottedOptionsChanged(prev, next);
+        this.filterOptions();
+        this.selectFirstElement();
+    }
+
+    /**
+     * Triggers the `suggestionProps` command to notify the tiptap editor to select the option.
+     * The method is defined based on the `Listbox.clickHandler` implementation.
+     *
+     * @internal
+     */
+    public override clickHandler(e: MouseEvent): boolean {
+        const capturedElement = (e.target as HTMLElement).closest(
+            'option,[role=option]'
+        );
+
+        const capturedListOption = capturedElement as ListOption | null;
+
+        if (!capturedListOption || capturedListOption.disabled) {
+            return false;
+        }
+        this.activateMention({
+            href: capturedListOption.value,
+            displayName: capturedListOption.text
+        } as MentionDetail);
```

Comment:
avoid `as` casts
const mentionDetail: MentionDetail = {
    href: capturedListOption.value,
    displayName: capturedListOption.text
};
this.activateMention(mentionDetail);

Reply:
Removed unwanted casting.

Reply:
Updated